packages/core/src/plugins/index.ts

```diff
@@ -728,69 +730,166 @@ const _overridePlugin = (c: RnvContext, pluginsPath: string, dir: string) => {
                 if (fsLstatSync(ovDir).isDirectory()) {
                     logWarning('overrides.json: Directories not supported yet. specify path to actual file');
                 } else {
-                    overrideFileContents(ovDir, override, overridePath);
+                    overrideFileContents(ovDir, override, overridePath, k);
                 }
             }
         });
     }
 
     // const parentDest = path.join(dir, '..')
 };
+export const overrideFileContents = (
+    dest: string,
+    override: Record<string, string>,
+    overridePath = '',
+    fileKey: string
+) => {
+    const overrideDir = _ensureOverrideDirExists();
+    const appliedOverrideFilePath = path.join(overrideDir, 'applied_overrides.json');
+
+    const appliedOverrides = _readAppliedOverrides(appliedOverrideFilePath);
 
-export const overrideFileContents = (dest: string, override: Record<string, string>, overridePath = '') => {
     if (fsExistsSync(dest)) {
         let fileToFix = fsReadFileSync(dest).toString();
-
+        const { backupPath, isFirstRun } = _saveOriginalFile(dest, overrideDir);
+        const packageName = _getPackageName(dest, fileKey);
+        const previousOverride = appliedOverrides[packageName]?.[fileKey] || {};
+        const overridesChanged = JSON.stringify(previousOverride) !== JSON.stringify(override);
+        if (overridesChanged && !isFirstRun) {
```

Comment:
It seems like after reinstalling node_modules or updating a package, overrides don't get applied to the new clean node_modules files. And after changing an override, an outdated backup is restored and then overridden.